src/frontend.rs

```diff
@@ -165,13 +176,80 @@ impl fmt::Display for StatementArg {
     }
 }
 
+// TODO: Incorporate custom statements into this enum.
+/// Type encapsulating statements with their associated arguments.
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub struct Statement(pub NativeStatement, pub Vec<StatementArg>);
+pub enum Statement {
+    None,
+    ValueOf(AnchoredKey, Value),
+    Equal(AnchoredKey, AnchoredKey),
+    NotEqual(AnchoredKey, AnchoredKey),
+    Gt(AnchoredKey, AnchoredKey),
+    Lt(AnchoredKey, AnchoredKey),
+    Contains(AnchoredKey, AnchoredKey),
+    NotContains(AnchoredKey, AnchoredKey),
+    SumOf(AnchoredKey, AnchoredKey, AnchoredKey),
+    ProductOf(AnchoredKey, AnchoredKey, AnchoredKey),
+    MaxOf(AnchoredKey, AnchoredKey, AnchoredKey),
+}
+
+impl Statement {
+    pub fn code(&self) -> NativeStatement {
+        match self {
+            Self::None => NativeStatement::None,
+            Self::ValueOf(_, _) => NativeStatement::ValueOf,
+            Self::Equal(_, _) => NativeStatement::Equal,
+            Self::NotEqual(_, _) => NativeStatement::NotEqual,
+            Self::Gt(_, _) => NativeStatement::Gt,
+            Self::Lt(_, _) => NativeStatement::Lt,
+            Self::Contains(_, _) => NativeStatement::Contains,
+            Self::NotContains(_, _) => NativeStatement::NotContains,
+            Self::SumOf(_, _, _) => NativeStatement::SumOf,
+            Self::ProductOf(_, _, _) => NativeStatement::ProductOf,
+            Self::MaxOf(_, _, _) => NativeStatement::MaxOf,
+        }
+    }
+    pub fn args(&self) -> Vec<StatementArg> {
+        use StatementArg::*;
+        match self.clone() {
+            Self::None => vec![],
+            Self::ValueOf(ak, v) => vec![Key(ak), Literal(v)],
+            Self::Equal(ak1, ak2) => vec![Key(ak1), Key(ak2)],
+            Self::NotEqual(ak1, ak2) => vec![Key(ak1), Key(ak2)],
+            Self::Gt(ak1, ak2) => vec![Key(ak1), Key(ak2)],
+            Self::Lt(ak1, ak2) => vec![Key(ak1), Key(ak2)],
+            Self::Contains(ak1, ak2) => vec![Key(ak1), Key(ak2)],
+            Self::NotContains(ak1, ak2) => vec![Key(ak1), Key(ak2)],
+            Self::SumOf(ak1, ak2, ak3) => vec![Key(ak1), Key(ak2), Key(ak3)],
+            Self::ProductOf(ak1, ak2, ak3) => vec![Key(ak1), Key(ak2), Key(ak3)],
+            Self::MaxOf(ak1, ak2, ak3) => vec![Key(ak1), Key(ak2), Key(ak3)],
+        }
+    }
+}
+
+impl Into<middleware::Statement> for Statement {
```

Comment:
It's preferable to implement the `From` trait so that we get both `From` and `Into`.
Rust allows implementing a `From` trait on a foreign type without issues.